### PR TITLE
Vehicle mounted autoclave

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
 ```SUMMARY: Content "Add autoclave vehicle part"```
#### Purpose of change
Autoclave can be turned into mountable autoclave using cable and power supply and then bolted unto a frame to be powered by the vehicle power grid.

#### Describe the solution
Add Autoclave as a vehicle part

#### Additional context
Autoclave still consume a lot of power
The recipe is autolearned

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -4485,6 +4485,29 @@ static void process_vehicle_items( vehicle &cur_veh, int part )
...
+                autoclave_finished = true;
+                cur_veh.parts[part].enabled = false;
+            } else if( calendar::once_every( 15_minutes ) ) {
+                add_msg( _( "It should take %d minutes to finish sterilising items in the %s." ),
+                         to_minutes<int>( time_left ) + 1, cur_veh.name );
+                break;
+            }
+        }
+        if( autoclave_finished ) {
+            add_msg( _( "The washing machine in the %s has finished washing." ), cur_veh.name );
```
Comment (Night-Pryanik) [inline src/map.cpp]:
```suggestion
            add_msg( _( "The autoclave in the %s has finished washing." ), cur_veh.name );
```

File: src/vehicle_use.cpp [hunk 2]
```diff
@@ -1353,6 +1353,48 @@ void vehicle::open_or_close( const int part_index, const bool opening )
...
+    bool cbms = std::all_of( items.begin(), items.end(), []( const item & i ) {
+        return i.is_bionic();
+    } );
+
+    if( parts[p].enabled ) {
+        parts[p].enabled = false;
+        add_msg( m_bad,
+                 _( "You turn the autoclave off before it's finished the program, and open its door." ) );
+    } else if( fuel_left( "water" ) < 8 && fuel_left( "water_clean" ) < 8 ) {
+        add_msg( m_bad, _( "You need 8 charges of water in tanks of the %s for the autodoc to run." ),
```
Comment (Night-Pryanik) [inline src/vehicle_use.cpp]:
```suggestion
        add_msg( m_bad, _( "You need 8 charges of water in tanks of the %s for the autoclave to run." ),
```

File: src/vehicle_use.cpp [hunk 3]
```diff
@@ -1353,6 +1353,48 @@ void vehicle::open_or_close( const int part_index, const bool opening )
...
+                 name );
+    } else if( filthy_items ) {
+        add_msg( m_bad,
+                 _( "You need to remove all filthy items from the autoclave to start the sterilising cycle." ) );
```
Comment (Night-Pryanik) [inline src/vehicle_use.cpp]:
```suggestion
                 _( "You need to remove all filthy items from the autoclave to start the sterilizing cycle." ) );
```

**Comment by Night-Pryanik** [comment]
Hmm, I've seen a code almost identical to this somewhere... :wink:

**Comment by Fris0uman** [comment]
I have no idea what you re talking about!
But autoclave and dishwasher are not copy-pasted from washing machine that's for sure

**Comment by Fris0uman** [comment]
I have no idea what you re talking about!
But autoclave and dishwasher are not copy-pasted from washing machine that's for sure

**Pull request merged**